src/Vlingo.Actors/Plugin/Mailbox/SharedRingBuffer/RingBufferDispatcher.cs

```diff
@@ -38,12 +40,15 @@ public void Execute(IMailbox mailbox)
 
         public void Start()
         {
-            if (started != null)
+            lock (mutex)
```

Comment:
While this is what the original implementation was, can I suggest a little bit better implementation? Double checking with a lock in the middle is a somewhat efficient implementation of a singleton. it would look something like-
```csharp
if (started == null)
{
  lock (mutext)
  {
    if (started == null)
      started = ... // initialize with task
  }
}
```

Reply:
This won't be thread-safe unless started is made `volatile`. As John Skeet stated:

> if I'm in a situation where I have to choose between double-checked locking and "lock every time" code I'd go for locking every time until I'd got real evidence that it was causing a bottleneck. When it comes to threading, a simple and obviously-correct pattern is worth a lot.

Thera are reasons for that it should be avoided.

So as this code just ensure that multiple threads should not start the same instance of actor if client messes up, I would just not make it more complicated as it should be. The performance in this case is not what this is striving for.